ci: allow debugging of doc workflows

nashif opened the pull request:
- ci: doc: upload build log for debugging
- ci: doc-build: allow manual invocation of workflow

nashif committed:
ci: doc: upload build log for debugging

When PDF build fails, we want to see what cmaused the failure, so upload
the log for later analysis.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

fabiobaltieri commented:
Could this be pushed on https://github.com/zephyrproject-rtos/zephyr-testing/ instead?

nashif committed:
ci: doc-build: set shell as bash

Specify the shell variant being used in the workflow step.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: doc-build: use customer runner

Apparently the PDF build needs more resources, otherwise we end up
aborting the build too early.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
